lattice_controller_engine/lib/lattice_controller_engine/reconciler/app_spec.ex

```diff
@@ -0,0 +1,20 @@
+defmodule LatticeControllerEngine.Reconciler.AppSpec do
+  alias LatticeControllerEngine.Reconciler.Traits
+
+  def reconcile(
+        desired = %LatticeControllerEngine.Model.AppSpec{},
+        actual = %LatticeControllerEngine.Observed.Lattice{}
+      ) do
+    for component <- desired.components,
+        trait <- component.traits do
+      Traits.reconcile_trait(desired, component, trait, actual)
+    end
+    |> List.flatten()
+  end
+
+  def matching_hosts(actual = %LatticeControllerEngine.Observed.Lattice{}, requirements = %{}) do
+    actual.hosts
+    |> Enum.filter(fn {_host_id, host} -> host.labels == requirements end)
+    |> Enum.map(fn {host_id, _host} -> host_id end)
+  end
```

Comment:
Are host labels required to be an exact match, or do we accept host labels that are a superset? e.g. if I need a host with `hostcore.arch == x86_64`, that should return a match even if that host has more labels

Reply:
D'OH.
Great catch here.